### PR TITLE
846 adjust naming and assignment of abm age groups

### LINKED ISSUE #846: adjust naming and assignment of ABM age groups
### Motivation / Current Behaviour

We want to apply recent changes to age groups in abm_minimal.cpp to all other ABM files.

### Enhancement description

The proposed changes include
- renaming age group variables to use lower-case letters, so they do not look like macros,
- changing value assignment from `AgeGroup(num_age_groups - n)` to simply `AgeGroup(n)`.
  - The idea for the former was that it could help a user assigning AgeGroups values in [0, num_age_groups). This is better achieved using documentation and some asserts in the ABM code.

### Additional context

_No response_

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

### PR DESCRIPTION
**Pull request by khoanguyen-dev**
# Changes and Information

Please **briefly list the changes** made, additional Information and what the Reviewer should look out for:

- Renaming age group variables to use lower-case letters, so they do not look like macros.
- Changing value assignment from AgeGroup(num_age_groups - n) to simply AgeGroup(n).
- Add check to functions that use AgeGroup as parameters

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [x] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development)

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)

closes #846

### CODE REVIEW

**Review by xsaschako — APPROVED** [review]
Looks good!

**Review by reneSchm — CHANGES_REQUESTED** [review]
The name changes are great, but the asserts need to be fixed. I added suggestions only to a few, since several are very similar.

**Comment by xsaschako** [comment]
I opened #873 and will merge this asap with rene

File: cpp/models/abm/testing_strategy.cpp [hunk 1]
```diff
@@ -43,6 +43,7 @@ bool TestingCriteria::operator==(const TestingCriteria& other) const
 
 void TestingCriteria::add_age_group(const AgeGroup age_group)
 {
+
```
[the remark this answers is not in the document]
Comment (xsaschako) [reply cpp/models/abm/testing_strategy.cpp:46]: true

**Pull request merged**